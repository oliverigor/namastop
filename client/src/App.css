.App {
  text-align: center;
  background-color: #eceff1;
  height: 100%;
}

.App-header {
  background-color: #1f6bff;
  min-height: 50%;
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: center;
  color: white;
  padding-bottom: 10px;
}
