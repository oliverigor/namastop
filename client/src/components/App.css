.App {
  text-align: center;
  background-color: #eceff1;
  height: 100%;
}

.App-header {
  z-index: 9999;
  background-color: #1f6bff;
  padding: 20px;
  top: 0;
  color: white;
  text-align: left;
  position: fixed;
  width: 100%;
}

.container-main {
  margin-left: 10%;
  margin-right: 10%;
  padding-top: 100px;
  font-size: large;
  padding-bottom: 100px;
  min-height: 100vh;
}

.card-image-override {
  border-radius: 50px;
  display: flex;
  justify-content: center;
}

.image-override {
  border-radius: 50%;
  width: 100px;
  height: 100px;
}

.message-text {
  font-style: italic;
  color: #35383c;
}
