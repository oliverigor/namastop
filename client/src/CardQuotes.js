import React, { Component } from "react";
import { Card, Col } from "react-materialize";
import "./App.css";
import logoNovatics from "./Marca_Novatics_negativo.png";

class CardQuotes extends Component {
	render() {
		return (
			<Col m={4} s={12}>
				<Card className="small" textClassName="black-text align-left">
					<div className="card-title">
						{this.props.quotes.name.first}
					</div>
					<div className="card-image">
						<img src={logoNovatics} alt="foto-bonita" />
						<div className="card-stacked">
							<div className="card-content">
								<p>{this.props.quotes.greeting}</p>
							</div>
						</div>
					</div>
				</Card>
			</Col>
		);
	}
}

export default CardQuotes;
