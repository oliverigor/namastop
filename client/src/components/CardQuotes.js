import React, { Component } from "react";
import { Card, Col } from "react-materialize";
import "./App.css";
import logoNovatics from "../images/Marca_Novatics_negativo.png";

class CardQuotes extends Component {
	render() {
		return (
			<Col m={4} s={12}>
				<Card className="small" textClassName="black-text align-left">
					<div className="card-image-override">
						<img
							src={logoNovatics}
							alt="foto-bonita"
							className="image-override"
						/>
					</div>
					<div className="card-title">{this.props.quotes.user}</div>

					<div className="card-stacked">
						<div className="card-content">
							<p>{this.props.quotes.message}</p>
						</div>
					</div>
				</Card>
			</Col>
		);
	}
}

export default CardQuotes;
