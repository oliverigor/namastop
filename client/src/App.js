import React, { Component } from "react";
import logoNovatics from "./Marca_Novatics_negativo.png";
import { Container, Row, Button } from "react-materialize";
import "./App.css";
import jsonData from "./quotes";
import firebase from "./firebase";

import CardQuotes from "./CardQuotes";

class App extends Component {
  state = {
    response: "",
    posts: "",
    responseToPost: "",
    data: jsonData,
    dataAPI: null
  };

  componentDidMount() {
    this.callApi()
      .then(res => this.setState({ response: res.express }))
      .catch(err => console.log(err));

    const oldPosts = firebase.database().ref("post");
    oldPosts.on(
      "value",
      snapshot => {
        let newState = [];
        var oldPosts = snapshot.val();
        for (let i in oldPosts) {
          oldPosts[i].map(aqui => {
            newState.push({
              id: aqui.id,
              message: aqui.message,
              user: aqui.user
            });
            return this.setState({
              posts: newState,
              dataAPI: newState
            });
          });
        }
      },
      function(error) {
        console.log("Error: " + error.code);
      }
    );
    // console.log({ oldPosts: oldPosts });
  }

  callApi = async () => {
    const response = await fetch("/api/hello");
    const body = await response.json();

    if (response.status !== 200) throw Error(body.message);

    return body;
  };

  handleNewQuotes = async () => {
    console.log("entrei aqui");
    const response = await fetch("/api/hello", {
      method: "GET",
      headers: {
        "Content-Type": "application/json"
      }
      // body: JSON.stringify({ posts: this.state.posts })
    });
    const body = await response.text();

    this.setState({ dataAPI: JSON.parse(body) });
    const post = this.state.dataAPI;
    const itemsRef = firebase.database().ref("post");
    itemsRef.push(post);
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <div>
            <img src={logoNovatics} alt="logo" height={30} />
          </div>
        </header>
        {/*<p>{this.state.response}</p>
        <form onSubmit={this.handleSubmit}>
          <p>
            <strong>Post to Server:</strong>
          </p>
          <input
            type="text"
            value={this.state.post}
            onChange={e => this.setState({ post: e.target.value })}
          />
          <button type="submit">Submit</button>
        </form>
        <p>{this.state.responseToPost}</p>*/}
        <div className="container-main">
          <Container>
            <Row>
              {this.state.dataAPI ? (
                this.state.dataAPI.map(quotes => (
                  <CardQuotes key={quotes.id} quotes={quotes} />
                ))
              ) : (
                <p>loading...</p>
              )}
            </Row>
            <Button
              floating
              large
              className="blue"
              waves="light"
              icon="add"
              onClick={() => this.handleNewQuotes()}
            />
          </Container>
        </div>
      </div>
    );
  }
}

export default App;
