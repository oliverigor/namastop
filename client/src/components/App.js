import React, { Component } from "react";
import logoNovatics from "../images/Marca_Novatics_negativo.png";
import { Container, Row, Button } from "react-materialize";
import "./App.css";

import CardQuotes from "./CardQuotes";

class App extends Component {
  state = {
    response: "",
    posts: "",
    responseToPost: "",
    dataA: null
  };

  componentDidMount() {
    this.callApi()
      .then(res => this.setState({ response: res.express }))
      .catch(err => console.log(err));
  }

  callApi = async () => {
    const response = await fetch("/api/posts");

    const body = await response.json();
    this.setState({ data: body });
    if (response.status !== 200) throw Error(body.message);

    return body;
  };

  handleNewQuotes = async () => {
    const response = await fetch("/api/posts", {
      method: "GET",
      headers: {
        "Content-Type": "application/json"
      }
    });
    const body = await response.json();

    this.setState({ data: body });
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <div>
            <img src={logoNovatics} alt="logo" height={30} />
          </div>
        </header>
        <div className="container-main">
          <Container>
            <Row>
              {this.state.data ? (
                this.state.data.map(quotes => (
                  <CardQuotes key={quotes.id} quotes={quotes} />
                ))
              ) : (
                <p>loading...</p>
              )}
            </Row>
            <Button
              large
              className="blue"
              waves="light"
              onClick={() => this.handleNewQuotes()}
            >
              {" "}
              + menções?
            </Button>
          </Container>
        </div>
      </div>
    );
  }
}

export default App;
