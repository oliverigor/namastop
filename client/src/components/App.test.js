import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { shallow } from "enzyme";
import ShallowRenderer from "react-test-renderer/shallow";
import CardQuotes from "./CardQuotes";

it("renders without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<App />, div);
});

test("Should render correctly", () => {
	const renderer = new ShallowRenderer();
	const quotes = { id: "HheuashaHh($*QR", user: "igor", message: "falavei" };
	const component = renderer.render(
		<CardQuotes key={quotes.id} quotes={quotes} />
	);

	expect(component).toMatchSnapshot();
});
