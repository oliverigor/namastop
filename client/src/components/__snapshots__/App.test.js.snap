// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<Col
  m={4}
  s={12}
>
  <Card
    className="small"
    textClassName="black-text align-left"
  >
    <div
      className="card-image-override"
    >
      <img
        alt="foto-bonita"
        className="image-override"
        src="Marca_Novatics_negativo.png"
      />
    </div>
    <div
      className="card-title"
    >
      igor
    </div>
    <div
      className="card-stacked"
    >
      <div
        className="card-content"
      >
        <p>
          falavei
        </p>
      </div>
    </div>
  </Card>
</Col>
`;
